ccsds/PacketTypes.asn1

```diff
@@ -24,26 +24,71 @@
 PacketTypes DEFINITIONS AUTOMATIC TAGS::= BEGIN
 EXPORTS ALL;
 IMPORTS
---    PUSC-UINT, PUSC-BOOL FROM BasicTypes      --needs to be added in JSON if uncommented
     ApplicationProcess-ID FROM ApplicationProcess;
 
+CCSDS-Packet {PacketPrimaryHeader, PacketDataField} ::= SEQUENCE
+{
+    packetPrimaryHeader PacketPrimaryHeader,
+    packetDataField PacketDataField
+}
+
+PacketPrimaryHeader {Packet-ID} ::= SEQUENCE
+{
+    packetVersionNumber PacketVersionNumberValue,
+    packet-ID Packet-ID,
+    packetSequenceControl PacketSequenceControl,
+    packetDataLength PacketDataLength
+}
 
-SecondaryHeaderFlag ::= BOOLEAN
+PacketVersionNumberValue ::= NULL
 
 Packet-ID {PacketType} ::= SEQUENCE
 {
     packetType PacketType,
-    secondaryHeaderFlag SecondaryHeaderFlag,
+--! secondaryHeaderFlag INTEGER (0 .. 1), --! moved to ACN
     applicationProcess-ID ApplicationProcess-ID
 }
 
+SecondaryHeaderFlag ::= INTEGER (0 .. 1)
+
 PacketSequenceControl ::= SEQUENCE
 {
     sequenceFlags NULL,
-    --packetSequenceCount PUSC-UINT (0 .. 16383)
-    packetSequenceCount INTEGER (0 .. 16383)
+    packetSequenceCountOrName INTEGER (0 .. 16383)
+}
+
+PacketDataLength ::= INTEGER (0 .. 65535)
+
+
+PacketDataField {SecondaryHeaderField, UserDataField, StandaloneUserDataField} ::= CHOICE
+{
+    dataWithSecondaryHeader SEQUENCE
+    {
+        secondaryHeader SecondaryHeaderField,
+        userData UserDataField
+    },
+    dataNoSecondaryHeader SEQUENCE
+    {
+        userData StandaloneUserDataField
+    }
+}
+
+UserDataField {Data} ::= SEQUENCE
+{
+    data Data,
+--! spare NULL,   --! defined as align-to-next acn encoding
+    packetErrorContol PacketErrorControl
+}
+
+PacketErrorControl ::= BIT STRING (SIZE(16))
+
+MessageType-ID ::= SEQUENCE
+{
+--! ServiceType-ID INTEGER (1 .. 255),
+--! MessageSubtype-ID INTEGER (1 .. 255) --! moved to ACN
```

Comment:
do not "nest" `--` comments - although asn1scc does not support it, but second `--` closes the comment, according to standard